_episodes/14.md

```diff
@@ -0,0 +1,330 @@
+process when running operations like list files on more modern object storage. 
+
+As all these problems amassed over the years, it was clear something needed
+to be done. In the last few years, a few candidate table formats have come to
+the forefront of data engineering trends. Apache Iceberg, Apache Hudi, and the
+proprietary Databricks' Deltalake. The goal of these systems was to modernize
+the old Hive data structure. To Trino, Iceberg was particularly promising due to
+the list of promising features like schema versioning support and hidden
+partitioning that made it particularly attractive. However, the biggest element
+that sets Iceberg apart is the specification they are writing around it. This
+not only enables people to understand how to use it, but documents how others
+can implement the same features with an entirely different systems. We will
+save a deep dive on this spec for hte next episode when we bring on Ryan Blue,
+creator of Iceberg, to dig into these details. Let's talk about Iceberg.
+
+### The Iceberg table format
+
+Iceberg, is a new table format developed at Netflix that aims to replace older 
+table formats like Hive to add better flexibility as the schema evolves, atomic
+operations, speed, and just dependability. To be clear, it's not a new file
+format, as it still uses ORC, Parquet, and Avro, but a table format. Netflix 
+donated Iceberg to the Apache Software Foundation and it is now a top level
+project! 
+
+Iceberg handles both the data on disk just like Hive, but instead it stores the
+metadata in manifest files on disk along with the data itself. These
+*manifest files* are metadata files that list a subset of data files that make
+up a snapshot. *Manifest lists* are a special type of manifest file that point
+to other manifest files. *Snapshots* contain a manifest list that points to all
+the manifest files that belong to the snapshot. Another huge difference from
+Hive is that the manifest files keep track of table data at the file level as
+opposed to directory level that Hive uses. By doing so, Iceberg avoids having to
+list all files in a directory, which becomes a very common and expensive
+operation.
+ 
+<p align="center">
+  <img align="center" width="60%" height="100%" src="/assets/episode/14/iceberg-metadata.png"/><br/>
+</p>
+ 
+By tracking files this way, we not only get begger performance from object
+storage, it also enables serializeable isolation. This addresses the lack of
+consistency between the metadata and file state experienced in Hive.
+
+One of the greater advantages to Iceberg over Hive is the in-place table
+evolution. This means that you can add, drop, or rename a column, as well as, 
+reorder and update a column without any expensive refactoring of tables or 
+moving data around and there will be no adverse effects on your data or
+metadata.
+
+Partition evolution and hidden partitions are particularly invaluable. Before
```

Comment:
Remove the sentence "Before in Hive, you had.." since this was already discussed earlier.